woocommerce/woocommerce-ios
# [Mobile Payments] Replace markIPPTransaction with newer stored dates

## Description


In #9366 we started to store the dates of first transactions per reader/site in the device storage.

This is a more detailed store than the set of siteIDs which previously backed `loadSiteHasAtLeastOneIPPTransactionFinished`, and they are saved at the same time.

It isn't necessary for us to keep saving both these sets of data.

This PR removes the method that stores the set of site IDs, and replaces the implementation of `loadSiteHasAtLeastOneIPPTransactionFinished` with one which checks the newer store instead. It also checks the old value where required, for backwards compatibility.

## Testing instructions


Using a store/device combination which has only had transactions before #9366 was merged

1. Put a breakpoint in `CardPresentPaymentsOnboardingIPPUsersRefresher.refreshIPPUsersOnboardingState()` on                 `self?.cardPresentPaymentsOnboardingUseCase.refresh()`.
2. Launch the app
3. Observe that the debugger stops on your breakpoint as expected.

Repeat the test using a store/device which has only been used for IPP transactions since #9366 was merged

1. Observe that the debugger still stops on your breakpoint as expected


## Screenshots



---
